app/components/CheckPointRedirect/CheckPointRedirect.tsx

```diff
@@ -0,0 +1,19 @@
+'use client'
+
+import { useEffect } from 'react'
+import { useRouter } from 'next/navigation'
+import { getCheckPoint } from '@/lib/progressSaving'
+
+
+export default function CheckpointRedirect() {
```

Comment:
I don't know if there is any other way to redirect without useEffect in this case.

I think @pavanydg has separated the code in a component because it requires the client side hook useEffect(), while our home page is a server side component. ([When to use Server and Client Components?](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#when-to-use-server-and-client-components))